apps/omg_child_chain/lib/omg_child_chain/fee_server.ex

```diff
@@ -56,34 +62,45 @@ defmodule OMG.ChildChain.FeeServer do
     {:ok, load_fees()}
   end
 
-  @doc """
-   Reads fee specification file if needed and updates :ets state with current fees information
-  """
+  def handle_info(:update_fee_spec, state) do
+    _ =
+      if Application.get_env(:omg_child_chain, :ignore_fees) do
+        _ = Logger.warn("Fee specs from file are ignored. Updates takes no effect.")
+      else
+        alarm = {:invalid_fee_file, Node.self(), __MODULE__}
+        _ = Alarm.set(alarm)
+
+        with :ok <- update_fee_spec(),
+             do: _ = Alarm.clear(alarm)
+      end
+
+    {:noreply, state}
+  end
+
+  # Reads fee specification file if needed and updates :ets state with current fees information
   @spec update_fee_spec() :: :ok
-  def update_fee_spec do
+  defp update_fee_spec do
     path = get_fees()
 
-    :ok =
-      with {:reload, changed_at} <- should_load_file(path),
-           {:ok, content} <- File.read(path),
-           {:ok, specs} <- Fees.parse_file_content(content) do
-        :ok = save_fees(specs, changed_at)
-        _ = Logger.info("Reloaded #{inspect(Enum.count(specs))} fee specs from file, changed at #{inspect(changed_at)}")
-
+    with {:reload, changed_at} <- should_load_file(path),
+         {:ok, content} <- File.read(path),
+         {:ok, specs} <- Fees.parse_file_content(content) do
+      :ok = save_fees(specs, changed_at)
+      _ = Logger.info("Reloaded #{inspect(Enum.count(specs))} fee specs from file, changed at #{inspect(changed_at)}")
+      :ok
+    else
+      {:file_unchanged, _last_change_at} ->
         :ok
-      else
-        {:file_unchanged, _last_change_at} ->
-          :ok
 
-        {:error, :enoent} ->
-          _ = Logger.error("The fee specification file #{inspect(path)} not found.")
+      {:error, :enoent} ->
```

Comment:
are these cases handled any differently apart from logging?

Reply:
Well it also re-map error if file is missing, otherwise it looks like:
`⋅Unable to update fees from file. Reason: {:error, :enoent}⋅`

Not sure if it's clear enough about the error cause (:question: ) I can remove above and leave just generic case here